Week 4/Word Break/Solution.java

```diff
@@ -0,0 +1,83 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+public class Solution {
+    public boolean validDict(String s,String[] dict) {
+    	if(s==null||dict==null) throw new IllegalArgumentException();
```

Comment:
Good job setting up the helper method and using it to set up your helper data structure and do input validation - that's exactly what you'll want to do in your interviews.